program/src/error.rs

```diff
@@ -0,0 +1,33 @@
+//! Program error types
+
+#[cfg(not(target_os = "solana"))]
+use solana_program::message::AddressLoaderError;
+use spl_program_error::*;
+
+#[spl_program_error]
+pub enum AddressLookupError {
+    /// Attempted to lookup addresses from a table that does not exist
+    #[error("Attempted to lookup addresses from a table that does not exist")]
+    LookupTableAccountNotFound,
+    /// Attempted to lookup addresses from an account owned by the wrong program
+    #[error("Attempted to lookup addresses from an account owned by the wrong program")]
+    InvalidAccountOwner,
+    /// Attempted to lookup addresses from an invalid account
+    #[error("Attempted to lookup addresses from an invalid account")]
+    InvalidAccountData,
+    /// Address lookup contains an invalid index
+    #[error("Address lookup contains an invalid index")]
+    InvalidLookupIndex,
+}
+
+#[cfg(not(target_os = "solana"))]
+impl From<AddressLookupError> for AddressLoaderError {
+    fn from(err: AddressLookupError) -> Self {
+        match err {
+            AddressLookupError::LookupTableAccountNotFound => Self::LookupTableAccountNotFound,
+            AddressLookupError::InvalidAccountOwner => Self::InvalidAccountOwner,
+            AddressLookupError::InvalidAccountData => Self::InvalidAccountData,
+            AddressLookupError::InvalidLookupIndex => Self::InvalidLookupIndex,
+        }
+    }
+}
```

Comment:
Let me narrow down where this is used. I believe it's by the transaction processor.

We can't just put this in the SDK because it will then make the SDK depend on the program crate, which I thought we wanted to avoid.

Reply:
Hm, that is annoying...

It looks like it might be possible to only do this one place where it's needed, let's see how that goes! https://github.com/anza-xyz/agave/pull/141

Reply:
Thanks for rolling this in! I'll take care of the other one.